src/misc/mntent.c

```diff
@@ -20,6 +21,43 @@ int endmntent(FILE *f)
 	return 1;
 }
 
+static inline int decode1(char** in_buf, char** out_buf,  const char* from, const char to) {
+	if(strncmp(from, *in_buf, strlen(from)) == 0) {
+		*(*out_buf)++ = to;
+		*in_buf += strlen(from);
+		return 1;
+	}
+	return 0;
+}
+
+static inline char* decode(char* buf) {
```

Comment:
iirc musl generally doesn't use inline keyword but let's the compiler decide whether it needs to be inlined (static is sufficient)

Reply:
I can remove the `assert`.  I had asserts for two reason:

* other parts of musl _do_ use them, try `git grep -w assert` for an overview.
* if the condition that triggers the assert happens, we already get undefined behaviour.  A clear crash is probably the least bad thing that can happen.  Especially in security critical applications.